### PR TITLE
CAMEL-14387 - fix NPE when client error

### PR DESCRIPTION
**Pull request by jbdefard**

### CODE REVIEW

File: components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/streaming/SubscriptionHelper.java [hunk 1]
```diff
@@ -16,6 +16,16 @@
  */
 package org.apache.camel.component.salesforce.internal.streaming;
 
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
```
Comment (oscerd) [inline components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/streaming/SubscriptionHelper.java]: Please run the build with the sourcecheck profile enabled, the static part must be at the end of the imports listing.

File: components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/streaming/SubscriptionHelper.java [hunk 2]
```diff
@@ -99,50 +99,51 @@ public SubscriptionHelper(final SalesforceComponent component) throws Salesforce
...
+            this.handshakeListener = new ClientSessionChannel.MessageListener() {
+                @Override
+                public void onMessage(final ClientSessionChannel channel, final Message message) {
                     LOG.debug("[CHANNEL:META_HANDSHAKE]: {}", message);
 
                     if (!message.isSuccessful()) {
                         LOG.warn("Handshake failure: {}", message);
-                        handshakeError = (String)message.get(ERROR_FIELD);
-                        handshakeException = getFailure(message);
+                        SubscriptionHelper.this.handshakeError = (String)message.get(ERROR_FIELD);
```
Comment (omarsmak) [inline components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/streaming/SubscriptionHelper.java]: What is the reason to add `SubscriptionHelper.this` for these properties? IMHO it makes the code a bit chaotic, same goes for `this`.

Comment (jbdefard) [reply components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/streaming/SubscriptionHelper.java]: Sorry is my mistake... (default configuration of my IDE). I fix this.

Comment (jbdefard) [reply components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/streaming/SubscriptionHelper.java]: I commited a more correct patch

**Comment by omarsmak** [comment]
Merged. Thank you @jbdefard !

**Pull request merged**